VirtoCommerce.DynamicExpressionsModule.Data/Promotion/Rewards/RewardItemForEveryNumInGetOfRel.cs

```diff
@@ -0,0 +1,38 @@
+using VirtoCommerce.Domain.Common;
+using VirtoCommerce.Domain.Marketing.Model;
+
+namespace VirtoCommerce.DynamicExpressionsModule.Data.Promotion
+{
+    //For [] in every [] items of entry [] get [] % off no more than [] not to exceed $ []
+    public class RewardItemForEveryNumInGetOfRel : DynamicExpression, IRewardExpression
+    {
+        public decimal Amount { get; set; }
+        public ProductContainer Product { get; set; } = new ProductContainer();
+        public int ForNthQuantity { get; set; }
+        public int InEveryNthQuantity { get; set; }
+        public int ItemLimit { get; set; }
+        public decimal MaxLimit { get; set; }
+
+        #region IRewardExpression Members
+
+        public PromotionReward[] GetRewards()
+        {
+            var reward = new CatalogItemAmountReward();
```

Comment:
I suggest to do not increase complicity by adding the new method
just write this code
```
var result = base.GetRewards().FirstOrDefault();
if(result != null)
{
result.ConditionalProductId = ConditionalProduct?.ProductId;

}
return result;
```

Reply:
We have method GetRewards() returning PromotionReward[] array, so we would need a cast in child object. In that case we would have non testable in compile time agreement for base class return type.
I think strictly typed method is better, so the complexity is reasonable here. 
What do yo think?

Reply:
But you directly inherit from the class where you created  this return object!

Reply:
Yes, but if you change the type of created reward in base class, you probably will have an error (or undesired behavior if using "as") for the child in runtime. Here i could see implicit dependency, which could be eliminated using this additional method. I think it is safer.
But OK, i could change it.